packages/completer/src/handler.ts

```diff
@@ -59,6 +61,9 @@ export class CompletionHandler implements IDisposable {
     void,
```

Comment:
I am pretty torn up about what to do with this existing getter and setter.

Do we know if anyone is using them?

I am tempted to just add a comment that they are both deprecated, keep the signatures the same for now, returning a dummy connector in the getter. 

Then remove them in 3.0?

Reply:
I think the only instance of it being used in core is [here](https://github.com/jupyterlab/jupyterlab/pull/8080/files#diff-472ee5441f1ad3f53f4e4a22d97bf9dbR367).

I don't believe it's being used by the LSP extension.

Reply:
What if we used [tagged unions](https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions) by adding an optional property in `IDataConnector`? 

This way, new connectors can identify themselves, either via boolean or string identifier. If the identifier is present, we return the dummy connector, otherwise we return `this._connector` casted to keep the getter as is.

Reply:
That would be nice!